contracts/pcv/curve/StableSwapOperatorV1.sol

```diff
@@ -0,0 +1,264 @@
+            );
+            emit Deposit(msg.sender, _daiOut);
+        }
+
+        // burn remaining FEI
+        uint256 _feiBalance = _fei.balanceOf(address(this));
+        _fei.burn(_feiBalance);
+    }
+
+    /// @notice withdraw DAI from the deposit. This will remove liquidity in the
+    /// current pool's proportion, burn the FEI part of it, and then use the 3crv
+    /// to withdraw liquidity of the 3pool in DAI.
+    /// note: because of slippage & fees on withdraw, the amount of DAI out is
+    ///        not exactly the expected amountUnderlying provided as input.
+    /// @param amountUnderlying of tokens withdrawn (0 = all)
+    /// @param to the address to send PCV to (self = no transfer)
+    function withdraw(address to, uint256 amountUnderlying)
+        public
+        override
+        onlyPCVController
+        whenNotPaused
+    {
+
+        uint256 _3crvVirtualPrice = IStableSwap(_3pool).get_virtual_price();
+        // withdraw all
+        if (amountUnderlying == 0) {
+            uint256 _lpToWithdraw = IERC20(pool).balanceOf(address(this));
+            uint256[2] memory _minAmounts; // [0, 0]
+            IERC20(pool).approve(pool, _lpToWithdraw);
+            IStableSwap(pool).remove_liquidity(_lpToWithdraw, _minAmounts);
+        }
+        // partial withdraw
+        else {
+            uint256 _amount3crvToWithdraw = amountUnderlying * 1e18 / _3crvVirtualPrice;
+            uint256 _lpTotalSupply = IERC20(pool).totalSupply();
+            uint256 _lpToWithdraw = _lpTotalSupply * _amount3crvToWithdraw / IStableSwap(pool).balances(_3crvIndex);
+
+            uint256[2] memory _minAmounts; // [0, 0]
+            IERC20(pool).approve(pool, _lpToWithdraw);
+            IStableSwap(pool).remove_liquidity(_lpToWithdraw, _minAmounts);
+        }
+
+        // remove 3pool liquidity in DAI (3crv -> DAI)
+        uint256 _3crvBalance = IERC20(_3crv).balanceOf(address(this));
+        uint256 _daiBalanceBefore = IERC20(_dai).balanceOf(address(this));
+        IERC20(_3pool).approve(_3pool, _3crvBalance);
+        IStableSwap(_3pool).remove_liquidity_one_coin(
+          _3crvBalance,
+          0, // 0 = DAI token index
+          0 // minimum 0 DAI out (defensively checked below)
+        );
+
+        // check slippage
+        uint256 _daiBalanceAfter = IERC20(_dai).balanceOf(address(this));
+        uint256 _minDaiOut = amountUnderlying * (BASIS_POINTS_GRANULARITY - depositMaxSlippageBasisPoints) / BASIS_POINTS_GRANULARITY;
+        uint256 _daiBalanceWithdrawn = _daiBalanceAfter - _daiBalanceBefore;
+        require(_daiBalanceWithdrawn >= _minDaiOut, "StableSwwapOperatorV1: 3pool withdraw slippage too high");
+
+        // compute amount to withdraw
+        uint256 _amount = amountUnderlying;
+        // @dev: there is a small slippage on remove, so the amount of DAI balance
+        //       will be slightly smaller than the amountUnderlying if the contract
+        //       held no DAI before withdraw.
```

Comment:
Is it possible to approximate/overestimate the withdraw amount to make the output exact?

Reply:
I added an overshoot of 0.015% to account for fees in `withdraw_liquidity_one_coin()`